src/ucp/core/ucp_rkey.c

```diff
@@ -79,6 +102,15 @@ ucs_status_t ucp_ep_rkey_unpack(ucp_ep_h ep, void *rkey_buffer, ucp_rkey_h *rkey
 
     /* Read remote PD map */
     pd_map   = *(uint64_t*)p;
+
+    ucs_trace("unpacking rkey with pd_map 0x%"PRIx64, pd_map);
+
+    if (pd_map == 0) {
+        /* Dummy key return ok */
+        *rkey_p = &dummy_rkey;
```

Comment:
are you using dummy_rkey both as rkey handle, and as an rkey buffer?
maybe would be better to separate them (for example, allocate rkey_**buffer** as usual and not point to dummy_rkey)